crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -305,7 +307,14 @@ def disable_non_core_proposals
     end
 
     # and finally delete the roles itself
-    active_non_core_roles.each(&:destroy)
+    roles_to_disable.each do |role|
+      if role.name == "crowbar-config-default"
+        role.elements["crowbar-upgrade"] = []
+        role.save
+      else
+        role.destroy
```

Comment:
You mean `active_non_core_roles.each(&:destroy)` called on the original roles only, right, without crowbar-config-default and crowbar-core?
I thought deleting "crowbar-upgrade" role is save, we do not need it any more.